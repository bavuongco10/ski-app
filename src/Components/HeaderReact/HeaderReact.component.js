import React from 'react';
import styled, { keyframes } from 'styled-components';

import { Images } from '../../Themes';
import BuildInfo from '../BuildInfo/BuildInfo.component';

const rotate360 = keyframes`
   from { transform: rotate(0deg); }
   to { transform: rotate(360deg); }
`;

const AppHeader = styled.header`
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
  text-align: center;
`;

const AppTitle = styled.h1`
  font-size: 1.5em;
`;

const AppImage = styled.img`
  height: 80px;
  animation: ${rotate360} infinite 20s linear;
`;

const HeaderReact = () => (
  <AppHeader className="App-header">
    <AppImage src={Images.react_logo} className="App-logo" alt="logo" />
    <AppTitle className="App-title">Welcome to React</AppTitle>
    <BuildInfo />
  </AppHeader>
);

export default HeaderReact;
