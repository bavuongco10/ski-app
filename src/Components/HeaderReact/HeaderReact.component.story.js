import React from 'react';
import { storiesOf } from '@storybook/react';

import HeaderReact from './HeaderReact.component';

storiesOf('HeaderReact', module).add('default', () => <HeaderReact />);
