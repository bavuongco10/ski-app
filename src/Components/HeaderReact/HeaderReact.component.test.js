import React from 'react';

import HeaderReact from './HeaderReact.component';

describe('HeaderReact component', () => {
  it('should renders without crashing', () => {
    const wrapper = shallow(<HeaderReact />);
    expect(wrapper).toMatchSnapshot();
  });
});
