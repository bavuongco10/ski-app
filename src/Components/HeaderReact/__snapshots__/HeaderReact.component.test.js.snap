// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderReact component should renders without crashing 1`] = `
<styled.header
  className="App-header"
>
  <styled.img
    alt="logo"
    className="App-logo"
    src="logo.svg"
  />
  <styled.h1
    className="App-title"
  >
    Welcome to React
  </styled.h1>
  <BuildInfo
    buildText="SHA: undefined | By: undefined"
    isProdEnv={false}
  />
</styled.header>
`;
